### PR TITLE
MINOR: Trivial Cleanups

### PR DESCRIPTION
**Pull request by dongjinleekr**
Remove unused variables, methods, parameters, unthrown exceptions, and fixing typos.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by dongjinleekr** [comment]
Here is some explanation about the last commit.

`ProducerRunnable#shutdown` method is never called, and the variable with the same name also not changed from its default value, `false`.

Furthermore, this variable does not work as intended; From the non-main thread's context, the modification of this variable's state may not be visible. To make it work as intended, it should be a volatile variable.

Since this variable is not used anymore, I thought removing it would be better.

**Comment by dongjinleekr** [comment]
@chia7712 @dengziming Could you have a look? This PR includes several trivial fixes I met reading the code, but `TestRecordTest#testProducerRecord`'s case is a little bit serious.

Since the types of `ProducerRecord` and `TestRecord` are different, calling `assertNotEquals` for these instances always succeeds.

File: streams/test-utils/src/test/java/org/apache/kafka/streams/test/TestRecordTest.java [hunk 1]
```diff
@@ -166,6 +167,9 @@ public void testProducerRecord() {
         final TestRecord<String, Integer> testRecord = new TestRecord<>(producerRecord);
         final TestRecord<String, Integer> expectedRecord = new TestRecord<>(key, value, headers, recordTime);
         assertEquals(expectedRecord, testRecord);
-        assertNotEquals(expectedRecord, producerRecord);
+        assertFalse(expectedRecord.getHeaders() == producerRecord.headers() &&
```
Comment (chia7712) [inline streams/test-utils/src/test/java/org/apache/kafka/streams/test/TestRecordTest.java]: Could we address this issue in separate PR? It seem to me this issue does not belong to "trivial cleanups" :)

Comment (dongjinleekr) [reply streams/test-utils/src/test/java/org/apache/kafka/streams/test/TestRecordTest.java]: @chia7712 I see. Let me open a separate PR. :smile:

**Comment by dongjinleekr** [comment]
@chia7712 Separated the testing issue and added some additional trivial cleanups.

@hachikuji the last commit on `NetworkChannel` is related to your commit on KAFKA-10842, which deleted `NetworkChannel#receive` but omitted to update the Javadoc accordingly. Is this what you meant?

**Comment by dongjinleekr** [comment]
ping @chia7712

**Comment by dongjinleekr** [comment]
Rebased onto the latest trunk. cc/ @chia7712

**Comment by dongjinleekr** [comment]
@chia7712 Ready to merge!

**Review by chia7712 — APPROVED** [review]
@dongjinleekr thanks for this patch. LGTM :)

**Pull request merged**